packages/@sanity/form-builder/src/inputs/TextInput.tsx

```diff
@@ -26,13 +27,34 @@ const TextInput = React.forwardRef(function TextInput(
     [markers]
   )
 
-  const handleChange = React.useCallback(
-    (event) => {
+  const [localValue, setlocalValue] = useState(null)
+  const inputRef = useForwardedRef(forwardedRef)
+
+  const throttledSubmit = useMemo(
+    () =>
+      throttle(
+        () => {
+          const inputValue = inputRef.current.value
+          onChange(PatchEvent.from(inputValue ? set(inputValue) : unset()))
+          setlocalValue(null)
+        },
+        500,
+        {leading: false, trailing: true}
+      ),
+    [inputRef, onChange]
+  )
+
+  const handleChange = useCallback(
+    (event: React.ChangeEvent<HTMLTextAreaElement>) => {
       const inputValue = event.currentTarget.value
-      onChange(PatchEvent.from(inputValue ? set(inputValue) : unset()))
+      setlocalValue(inputValue)
+      throttledSubmit()
     },
-    [onChange]
+    [throttledSubmit]
   )
+
+  useEffect(() => () => throttledSubmit(), [throttledSubmit])
```

Comment:
Same as above, should be `throttledSubmit.flush()` here as well I believe

Reply:
Yeah